processor/logdeduplicationprocessor/counter.go

```diff
@@ -0,0 +1,165 @@
+			lr := scopeLogs.LogRecords().AppendEmpty()
+
+			baseRecord := lc.logRecord
+
+			// Copy contents of base record
+			baseRecord.SetObservedTimestamp(pcommon.NewTimestampFromTime(lc.firstObservedTimestamp))
+			baseRecord.Body().CopyTo(lr.Body())
+
+			lr.Attributes().EnsureCapacity(baseRecord.Attributes().Len())
+			baseRecord.Attributes().CopyTo(lr.Attributes())
+
+			lr.SetSeverityNumber(baseRecord.SeverityNumber())
+			lr.SetSeverityText(baseRecord.SeverityText())
+
+			// Add attributes for log count and timestamps
+			lr.SetObservedTimestamp(pcommon.NewTimestampFromTime(lc.firstObservedTimestamp))
+			lr.SetTimestamp(pcommon.NewTimestampFromTime(timeNow()))
+			lr.Attributes().PutInt(l.logCountAttribute, lc.count)
+
+			firstTimestampStr := lc.firstObservedTimestamp.In(l.timezone).Format(time.RFC3339)
+			lastTimestampStr := lc.lastObservedTimestamp.In(l.timezone).Format(time.RFC3339)
+			lr.Attributes().PutStr(firstObservedTSAttr, firstTimestampStr)
+			lr.Attributes().PutStr(lastObservedTSAttr, lastTimestampStr)
+		}
+	}
+
+	return logs
+}
+
+// Add adds the logRecord to the resource aggregator that is identified by the resource attributes
+func (l *logAggregator) Add(resourceAttrs pcommon.Map, logRecord plog.LogRecord) {
+	key := pdatautil.MapHash(resourceAttrs)
+	resourceCounter, ok := l.resources[key]
+	if !ok {
+		resourceCounter = newResourceAggregator(resourceAttrs)
+		l.resources[key] = resourceCounter
+	}
+
+	resourceCounter.Add(logRecord)
+}
+
+// Reset resets the counter.
+func (l *logAggregator) Reset() {
+	l.resources = make(map[[16]byte]*resourceAggregator)
+}
+
+// resourceAggregator dimensions the counter by resource.
+type resourceAggregator struct {
+	attributes  pcommon.Map
+	logCounters map[[8]byte]*logCounter
+}
+
+// newResourceAggregator creates a new ResourceCounter.
+func newResourceAggregator(attributes pcommon.Map) *resourceAggregator {
+	return &resourceAggregator{
+		attributes:  attributes,
+		logCounters: make(map[[8]byte]*logCounter),
+	}
+}
+
+// Add increments the counter that the logRecord matches.
+func (r *resourceAggregator) Add(logRecord plog.LogRecord) {
+	key := getLogKey(logRecord)
+	lc, ok := r.logCounters[key]
+	if !ok {
+		lc = newLogCounter(logRecord)
+		lc.firstObservedTimestamp = timeNow().UTC()
+		r.logCounters[key] = lc
+	}
+	lc.Increment()
+}
+
+// logCounter is a counter for a log record.
+type logCounter struct {
+	logRecord              plog.LogRecord
+	firstObservedTimestamp time.Time
+	lastObservedTimestamp  time.Time
+	count                  int64
+}
+
+// newLogCounter creates a new AttributeCounter.
+func newLogCounter(logRecord plog.LogRecord) *logCounter {
+	return &logCounter{
+		logRecord: logRecord,
+		count:     0,
+	}
+}
+
+// Increment increments the counter.
+func (a *logCounter) Increment() {
+	a.lastObservedTimestamp = timeNow().UTC()
+	a.count++
+}
+
+// getLogKey creates a unique md5 hash for the log record to use as a map key
```

Comment:
```suggestion
// getLogKey creates a unique hash for the log record to use as a map key
```
Since it appears we're using fnv